libraries/AP_InertialSensor/AP_InertialSensor_RST.cpp

```diff
@@ -0,0 +1,401 @@
+ */
+/*
+  IMU driver for Robsense PhenixPro Devkit board including i3g4250d and iis328dq
+ */
+#include <AP_HAL/AP_HAL.h>
+
+#if CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RST_ZYNQ
+#include "AP_InertialSensor_RST.h"
+#include <AP_Math/AP_Math.h>
+
+#include <inttypes.h>
+#include <utility>
+#include <math.h>
+#include <stdio.h>
+
+const extern AP_HAL::HAL &hal;
+
+#define ADDR_INCREMENT                (1<<6)
+
+/************************************iis328dq register addresses *******************************************/
+#define ACCEL_WHO_AM_I             0x0F
+#define ACCEL_WHO_I_AM             0x32
+
+#define ACCEL_CTRL_REG1            0x20
+/* keep lowpass low to avoid noise issues */
+#define RATE_50HZ_LP_37HZ          (0<<4) | (0<<3)
+#define RATE_100HZ_LP_74HZ         (0<<4) | (1<<3)
+#define RATE_400HZ_LP_292HZ        (1<<4) | (0<<3)
+#define RATE_1000HZ_LP_780HZ       (1<<4) | (1<<3)
+
+#define ACCEL_CTRL_REG2            0x21
+#define ACCEL_CTRL_REG3            0x22
+#define ACCEL_CTRL_REG4            0x23
+
+#define ACCEL_CTRL_REG5            0x24
+#define ACCEL_HP_FILTER_RESETE     0x25
+#define ACCEL_OUT_REFERENCE        0x26
+#define ACCEL_STATUS_REG           0x27
+#define ACCEL_OUT_X_L              0x28
+#define ACCEL_OUT_X_H              0x29
+#define ACCEL_OUT_Y_L              0x2A
+#define ACCEL_OUT_Y_H              0x2B
+#define ACCEL_OUT_Z_L              0x2C
+#define ACCEL_OUT_Z_H              0x2D
+#define ACCEL_INT1_CFG             0x30
+#define ACCEL_INT1_SRC             0x31
+#define ACCEL_INT1_TSH             0x32
+#define ACCEL_INT1_DURATION        0x33
+#define ACCEL_INT2_CFG             0x34
+#define ACCEL_INT2_SRC             0x35
+#define ACCEL_INT2_TSH             0x36
+#define ACCEL_INT2_DURATION        0x37
+
+
+/* Internal configuration values */
+#define ACCEL_REG1_POWER_NORMAL    ((0<<7) | (0<<6) | (1<<5))
+#define ACCEL_REG1_Z_ENABLE        (1<<2)
+#define ACCEL_REG1_Y_ENABLE        (1<<1)
+#define ACCEL_REG1_X_ENABLE        (1<<0)
+
+#define ACCEL_REG4_BDU             (1<<7)
+#define ACCEL_REG4_BLE             (1<<6)
+#define ACCEL_REG4_FULL_SCALE_BITS ((1<<5) | (1<<4))
+#define ACCEL_REG4_FULL_SCALE_2G   ((0<<5) | (0<<4))
+#define ACCEL_REG4_FULL_SCALE_4G   ((0<<5) | (1<<4))
+#define ACCEL_REG4_FULL_SCALE_8G   ((1<<5) | (1<<4))
+
+#define ACCEL_STATUS_ZYXOR         (1<<7)
+#define ACCEL_STATUS_ZOR           (1<<6)
+#define ACCEL_STATUS_YOR           (1<<5)
+#define ACCEL_STATUS_XOR           (1<<4)
+#define ACCEL_STATUS_ZYXDA         (1<<3)
+#define ACCEL_STATUS_ZDA           (1<<2)
+#define ACCEL_STATUS_YDA           (1<<1)
+#define ACCEL_STATUS_XDA           (1<<0)
+
+#define ACCEL_DEFAULT_RANGE_G               8
+#define ACCEL_DEFAULT_RATE                  1000
+#define ACCEL_DEFAULT_ONCHIP_FILTER_FREQ    780
+#define ACCEL_ONE_G                         9.80665f
+
+/************************************i3g4250d register addresses *******************************************/
```

Comment:
this seems to be a standalone sensor and the implementation seem to be completely independent of them being together (i.e. not even the bus callback is shared). This is not like a sensor inside a package as is the case for MPU*. I think it would better to have gyro and accel on different classes so in future it would be easier to mix and match them with other sensor.  We can live with this as is for now, though.

Reply:
OK, thank you for your advice, I will separate them when I have time